pygfx/utils/text/_fontfinder.py

```diff
@@ -0,0 +1,260 @@
+            info = {"mtime": os.path.getmtime(p)}
+            dirs[p.replace("\\", "/")] = info
+        files = cache["files"]
+        for p in new_file_paths:
+            info = {"mtime": os.path.getmtime(p)}
+            files[p.replace("\\", "/")] = info
+
+        # Sort the keys, just being clean
+        cache = {"dirs": {}, "files": {}}
+        for key in sorted(dirs.keys()):
+            cache["dirs"][key] = dirs[key]
+        for key in sorted(files.keys()):
+            cache["files"][key] = files[key]
+
+        # Write to file system, first to a uniquely named file
+        filename2 = filename + ".part." + secrets.token_urlsafe(4)
+        with open(filename2, "wt", encoding="utf-8") as f:
+            json.dump(cache, f)
+
+        # Now we must replace the font cache on disk with the result.
+        # But we must take into account that another process might be
+        # reading from it. The os.replace() call is atomic. And on Unix
+        # that's that - whatever is reading is can keep doing so,
+        # because this is simply a new inode with the same name. On
+        # Windows, however, the replace will fail if something else has
+        # the file open. So we wait and try again, until we time out
+        # and give up.
+        etime = time.time() + 10
+        while time.time() < etime:
+            try:
+                os.replace(filename2, filename)
+            except OSError:
+                time.sleep(0.1)
+            else:
+                break
+        else:
+            logger.error("Failed to write font cache")
+            try:
+                os.remove(filename2)
+            except Exception:
+                pass
+
+    return list(cache["files"].keys())
+
+
+def find_system_fonts():
+    """Find fonts on the file system. Returns (dir_paths, file_paths).
+    This function might be relatively slow.
+    """
+    dirs = get_system_font_directories()
+    dir_paths = set()
+    file_paths = set()
+    for d in dirs:
+        new_dirs, new_files = find_fonts_from_dir(d, True)
+        dir_paths.update(new_dirs)
+        file_paths.update(new_files)
+
+    return dir_paths, file_paths
+
+
+# %% OS-specific logic
+
+
+def get_system_font_directories():
+    # Simple triage, easy to replace in tests.
+    if sys.platform.startswith("win"):
+        return get_windows_font_directories()
+    elif sys.platform.startswith("darwin"):
+        return get_osx_font_directories()
+    else:
+        return get_unix_font_directories()
+
+
+def get_windows_font_directories():
+    import winreg
+
+    dirs = []
+    dirs.extend(WinFontDirs)
+
+    # Get win32 font directory from the registry
+    ms_folders = r"Software\Microsoft\Windows\CurrentVersion\Explorer\Shell Folders"
+    try:
+        with winreg.OpenKey(winreg.HKEY_CURRENT_USER, ms_folders) as user:
+            dirs.append(winreg.QueryValueEx(user, "Fonts")[0])
+    except OSError:
+        dirs.append(os.path.join(os.environ["WINDIR"], "Fonts"))
```

Comment:
why not add both of these dirs?

Reply:
Good point. Took this from MPL, but I see no reason not to add both.